cpp/include/raft/linalg/coalesced_reduction.cuh

```diff
@@ -70,6 +73,85 @@ void coalescedReduction(OutType* dots,
   detail::coalescedReduction(dots, data, D, N, init, stream, inplace, main_op, reduce_op, final_op);
 }
 
+/**
+ * @defgroup coalesced_reduction Coalesced Memory Access Reductions
+ * For reducing along rows for col-major and along columns for row-major
+ * @{
+ */
+
+/**
+ * @brief Compute reduction of the input matrix along the leading dimension
+ *
+ * @tparam InElementType the input data-type of underlying raft::matrix_view
+ * @tparam LayoutPolicy The layout of Input/Output (row or col major)
+ * @tparam OutElementType the output data-type of underlying raft::matrix_view and reduction
+ * @tparam IndexType Integer type used to for addressing
+ * @tparam MainLambda Unary lambda applied while acculumation (eg: L1 or L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*MainLambda)(InType, IdxType);</pre>
+ * @tparam ReduceLambda Binary lambda applied for reduction (eg: addition(+) for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*ReduceLambda)(OutType);</pre>
+ * @tparam FinalLambda the final lambda applied before STG (eg: Sqrt for L2 norm)
+ * It must be a 'callable' supporting the following input and output:
+ * <pre>OutType (*FinalLambda)(OutType);</pre>
+ * @param handle raft::handle_t
+ * @param dots Output of type raft::device_matrix_view
+ * @param data Input of type raft::device_matrix_view
+ * @param init initial value to use for the reduction
+ * @param main_op elementwise operation to apply before reduction
+ * @param reduce_op binary reduction operation
+ * @param final_op elementwise operation to apply before storing results
+ * @param inplace reduction result added inplace or overwrites old values?
+ */
+template <typename InElementType,
+          typename LayoutPolicy,
+          typename OutElementType = InElementType,
+          typename IndexType      = std::uint32_t,
+          typename MainLambda     = raft::Nop<InElementType>,
+          typename ReduceLambda   = raft::Sum<OutElementType>,
+          typename FinalLambda    = raft::Nop<OutElementType>>
+void coalesced_reduction(
+  const raft::handle_t& handle,
+  raft::device_matrix_view<OutElementType, IndexType, LayoutPolicy> dots,
+  const raft::device_matrix_view<InElementType, IndexType, LayoutPolicy> data,
+  OutElementType init,
+  bool inplace           = false,
```

Comment:
Would you consider making the "in-place" version a different function or overload, rather than specifying this by `bool` parameter?  This is the approach taken by stdblas (P1673) for distinguishing in-place and out-of-place operations.

Reply:
Looking at the code implementation, I'm not sure if `inplace` is doing exactly what the name suggests it should be doing